ssl/statem/statem_clnt.c

```diff
@@ -2422,11 +2423,11 @@ MSG_PROCESS_RETURN tls_process_key_exchange(SSL_CONNECTION *s, PACKET *pkt)
 MSG_PROCESS_RETURN tls_process_certificate_request(SSL_CONNECTION *s,
                                                    PACKET *pkt)
 {
-    size_t i;
-
-    /* Clear certificate validity flags */
-    for (i = 0; i < SSL_PKEY_NUM; i++)
-        s->s3.tmp.valid_flags[i] = 0;
+    /* Allocate and clear certificate validity flags */
+    OPENSSL_free(s->s3.tmp.valid_flags);
+    s->s3.tmp.valid_flags = OPENSSL_zalloc(s->ssl_pkey_num * sizeof(uint32_t));
```

Comment:
If `s->s3.tmp.valid_flags` is already allocated, wouldn't a simple memset suffice?

Reply:
Yes, so changed in 0ac814f8247d1a5f157eec2b62bce0000c3a2d1e.